### PR TITLE
Set up sitespeed.io CI job.

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Add a GitHub action to run sitespeed on merge.

### QA steps

- Check that the action artefact has the sitespeed results. Here's a passing run (I temporarily changed the action to run on this PR): https://github.com/canonical-web-and-design/maas-ui/actions/runs/2567167138

## Fixes

Fixes: https://github.com/canonical-web-and-design/app-tribe/issues/1076.

### CODE REVIEW

**Comment by huwshimi** [comment]
On hold until we can use sampledata to set up machines.

**Comment by Caleb-Ellis** [comment]
I think this is unblocked now. Instead of generating new sample data every time we run CI, the core team have a few Jenkins jobs for creating a db dump of 10, 100 and 1000 sample machines ([link](http://maas-ci.internal:8080/view/maas-sampledata-dumper-master/)), which should be able to be downloaded and imported into a local PostgreSQL server.

With `maas-test-db`, we should be able to do:
``` bash
sudo cp maasdb.dump /var/snap/maas-test-db/common
sudo snap run --shell maas-test-db.psql -c 'db-dump restore $SNAP_COMMON/maasdb.dump maassampledata'
```

**Comment by huwshimi** [comment]
@Caleb-Ellis @petermakowski I've got this is working with the sample data, however the database dump with 1000 machines is about 8mb. Where do you think we should store that? Maybe on the asset server?

**Comment by huwshimi** [comment]
This is ready for review now!

**Comment by petermakowski** [comment]
> This is ready for review now!

The detailed summary in the report indicates there were no JS, CSS or font files loaded. Either something is not right or I don't understand how it's supposed to work.

image

File: .github/workflows/sitespeed.yml
Comment (petermakowski) [inline .github/workflows/sitespeed.yml:4]: Could we run this action on opened PRs and identify performance issues before changes are merged?

File: .github/workflows/sitespeed.yml
Comment (petermakowski) [inline .github/workflows/sitespeed.yml:22]: Does this dump need to be updated manually or do we/could we automate this?

File: docs/HACKING.md
Comment (petermakowski) [inline docs/HACKING.md:617]: Would be great if we could just read the hostname and port values from the `MAAS_URL` env variable, we wouldn't need this extensive readme and differentiation between local and remote MAAS deployments.

File: sitespeed.io/login.js
Comment (petermakowski) [inline sitespeed.io/login.js:15]: This is nitpicking, but I wonder why you decided not to abstract this to a variable as it's used in 2 places, but did in the other file where a timeout is only used in one place.

File: sitespeed.io/scripts/machines.js
Comment (petermakowski) [inline sitespeed.io/scripts/machines.js:3]: This is only used in 1 place. Was your intention to make this clearer and improve the readability?

File: sitespeed.io/scripts/machines.js
Comment (petermakowski) [inline sitespeed.io/scripts/machines.js:9]: If I understand correctly, we're waiting an arbitrary amount of time before stopping the measurement which does not guarantee we're measuring the entire page load cycle (until it's fully loaded and interactive). 2 minutes is probably sufficient, but could you think of any other ways of checking the page has fully loaded?

**Review by petermakowski — COMMENTED** [review]
This looks good overall, I just have a few questions/comments. Would be also good to solve conflicts and verify the report is correct before merge (comment above).

**Review by huwshimi — COMMENTED** [review]
I figured out the caching issue. By default it clears the cache between tests, [but not when using a preScript](https://www.sitespeed.io/documentation/sitespeed.io/best-practice/#clear-browser-cache-between-runs) (which we use for logging in). I've updated the test to clear the cache before the test and also added a second test after caching.

It's looking better now:

Screen Shot 2022-07-01 at 1 51 49 pm

File: sitespeed.io/scripts/machines.js
Comment (petermakowski) [reply sitespeed.io/scripts/machines.js:18]: Oooh, I missed that "small" detail. This should be perfectly fine then.

**Review by petermakowski — APPROVED** [review]

**Pull request merged**
